pkg/adsc/adsc.go

```diff
@@ -616,28 +621,33 @@ func (a *ADSC) mcpToPilot(m *mcp.Resource) (*config.Config, error) {
 		},
 	}
 
-	if !config.ObjectInRevision(c, a.cfg.Revision) { // In case upstream does not support rev in node meta.
-		return nil, nil
+	matchRev := config.ObjectInRevision(c, rev)
+	if !matchRev && strict { // In case upstream does not support rev in node meta.
+		return nil, matchRev, nil
 	}
 
 	if c.Meta.Annotations == nil {
 		c.Meta.Annotations = make(map[string]string)
 	}
 	nsn := strings.Split(m.Metadata.Name, "/")
 	if len(nsn) != 2 {
-		return nil, fmt.Errorf("invalid name %s", m.Metadata.Name)
+		return nil, matchRev, fmt.Errorf("invalid name %s", m.Metadata.Name)
 	}
 	c.Namespace = nsn[0]
 	c.Name = nsn[1]
+	if !matchRev {
```

Comment:
why return early here?

Reply:
According to the adsc use case, a previously adopted resource is not written to the store when a subsequent update results in a rev mismatch. 
In delta mode, this means that the resource needs to be removed from the store, and all we need is its namespace and name.